src/compiler/checker.ts

```diff
@@ -11507,7 +11507,7 @@ namespace ts {
                 if (symbol === globalThisSymbol) {
                     const varsOnly = new Map<string, Symbol>() as SymbolTable;
                     members.forEach(p => {
-                        if (!(p.flags & SymbolFlags.BlockScoped)) {
+                        if (!(p.flags & SymbolFlags.BlockScoped) && !(p.flags & SymbolFlags.ValueModule && some(p.declarations, isAmbientModule))) {

```

Comment:
`some`? What if you make a global named `"module"` that ?merges? with the ambient module named `"module"`? (Is such a merge even possible?)

Reply:
How do you make a global named `"module"`?